qiskit/qobj/converters/pulse_instruction.py

```diff
@@ -345,8 +347,10 @@ def convert_frame_change(self, instruction):
             phase_expr = parse_string_expr(phase, partial_binding=False)
 
             def gen_fc_sched(*args, **kwargs):
-                phase = abs(phase_expr(*args, **kwargs))
-                return commands.FrameChange(phase)(channel) << t0
+                _phase = phase_expr(*args, **kwargs)
```

Comment:
Does this often return a complex value? If it does not we should remove the check and just let this error be caught at a later time by allow Python duck typing to work.

Reply:
I don't think so. Usually parser returns a real value when imaginary part is zero. It returns complex value only when backends intentionally assign complex value as a phase, but this is invalid configuration.

Reply:
We can do this in marshmallow as well.

Reply:
So what is the path forward do you think?

Reply:
I realized real value is guaranteed by marshmallow.
https://github.com/Qiskit/qiskit-terra/blob/27b35555f3342ec6bbdedacf7cc1e2daf58961fe/qiskit/qobj/models/pulse.py#L56

If a user try to bundle complex value with `cmd_def.get`, marshmallow will raise an error during qobj generation. So I can simply remove type checking here.

Reply:
Great!